RoddyCore/src/de/dkfz/roddy/execution/jobs/Job.groovy

```diff
@@ -333,6 +338,100 @@ class Job extends BEJob<BEJob, JobResult> {
         path
     }
 
+
+    /**
+     * Stores a new job jobState info to an execution contexts job jobState log file.
+     *
+     * @param job
+     */
+    void appendToJobStateLogfile(BatchEuphoriaJobManager jobManager, ExecutionContext executionContext, BEJobResult res, OutputStream out = null) {
+        // Ugly, but works. Java/Groovy's missing runtime dispatch on method arguments sucks.
+        if (jobManager instanceof ClusterJobManager) {
+            appendToJobStateLogfileForCluster(executionContext, res)
+        } else if (jobManager instanceof DirectSynchronousExecutionJobManager) {
+            appendToJobStateLogfileLocal(executionContext, res, out)
+        } else {
+            throw new NotImplementedException()
+        }
+    }
+    private static String jobStateInfoLine(String jobId, String code, String millis) {
+        return String.format("%s:%s:%s", jobId, code, millis)
+    }
+
+
+    void appendToJobStateLogfileForCluster(ExecutionContext executionContext, BEJobResult res) {
+        if (res.wasExecuted) {
+            def job = res.command.getJob()
+            String jobInfoLine = getJobStateInfoLineForClusterJob(job)
+            if (jobInfoLine != null)
+                FileSystemAccessProvider.getInstance().appendLineToFile(true, executionContext.getRuntimeService().getNameOfJobStateLogFile(executionContext), jobInfoLine, false)
+        }
+    }
+
+    static String getJobStateInfoLineForClusterJob(BEJob job) {
```

Comment:
Oh that took me some time to see what happens :D 
Maybe some comments would help?
Maybe getJobState... is the wrong name? AssembleJobStateInfoLine?